designs/2024-deprecated-rule-metadata/README.md

```diff
@@ -0,0 +1,268 @@
+```
+
+These properties are often used for generating plugin/rule documentation websites and in documentation tooling like [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator).
+
+But there are some limitations to this current format:
+
+- Simply providing the replacement rule name as a string doesn't yield much context/explanation of the replacement/deprecation. That means documentation tooling / websites and code editors can only generate limited information to present about the situation.
+- Some rules provide the replacement rule name with the plugin prefix as in `prefix/rule-name` while others just provide it as `rule-name`, which can result in ambiguity about whether the replacement rule is in the same plugin, a different third-party plugin, or ESLint core. And for third-party plugins, there's no easy way to automatically determine where their documentation is located or how to link to them.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+We propose extended `meta.deprecated` and `meta.replacedBy` rule property schemas to reduce ambiguity and allow additional key details to be represented, described below as a TypeScript type for clarity:
+
+```ts
+type RuleMeta = {
+  deprecated?:
+    | boolean // Existing boolean option, backwards compatible.
+    | string // Shorthand property for general deprecation message, such as why the deprecation occurred. Any truthy value implies deprecated.
+    | {
+        message?: string; // General deprecation message, such as why the deprecation occurred.
+        url?: string; // URL to more information about this deprecation in general.
+      }
+    | undefined;
+  replacedBy?:
+    | readonly string[] // Existing shorthand property for rule names, backwards compatible. It's recommended to omit the plugin prefix from rule names.
+    | readonly {
+        /**
+         * Plugin containing the replacement.
+         * Use "eslint" if the replacement is an ESLint core rule.
+         * Omit if the replacement is in the same plugin.
+         */
+        plugin?:
+          | string // Shorthand property for the plugin name i.e. "eslint-plugin-example" that contains the replacement rule.
+          | {
+              name?: string; // Plugin name i.e. "eslint-plugin-example" that contains the replacement rule.
+              url?: string; // URL to plugin documentation.
+            };
+        rule?:
+          | string // Shorthand property for replacement rule name (without plugin prefix).
+          | {
+              name?: string; // Replacement rule name (without plugin prefix).
+              url?: string; // URL to rule documentation.
+            };
+        meta?: {
```

Comment:
I think `meta.replacedBy[].meta` is a bit confusing. It seems like the purpose of this `meta` is really to provide additional info about the replacement. Perhaps `note` or `info` would be a more appropriate name?

Reply:
`info` sounds good to me.